src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManager.cs

```diff
@@ -226,6 +226,11 @@ private static OpenType GetOpenType(Type t)
                 }
             }
 
+            if (typeDest == typeof(JArray) && typeof(IEnumerable).IsAssignableFrom(typeSource))
```

Comment:
Ok, good point on too general (we don't want string).

However, I think T = JObject or JToken is too restrictive. If a user defines a builder to IEnumerable<CustomType> I think we should convert that without the extra JObject/JToken constraint.